module/os/windows/zfs/zfs_ioctl_os.c

```diff
@@ -129,8 +131,16 @@ NTSTATUS zpool_zfs_get_metrics(PDEVICE_OBJECT DeviceObject, PIRP Irp, PIO_STACK_
 	}
 
 	vdev_stat_t vs = { 0 };
+	unsigned __int64 mirror_slog_alloc_size = 0, mirror_slog_space = 0;
 	spa_config_enter(spa_perf, SCL_ALL, FTAG, RW_READER);
-	perf->mirror_slog_alloc_size = getMirrorSlogUsedSize(spa_perf->spa_root_vdev);
+	if (spa_perf->spa_l2cache.sav_vdevs[0]->vdev_isl2cache)
```

Comment:
Can you do this similar to mirror slog? Sum over all L2ARC and return. Besides accessing without null check might be problematic

Reply:
Does this give aggregate space of all the l2cache devices? 
Check for the array size. spa_perf->spa_l2cache.sav_count gives the size of the sav_vdevs[] array?

Reply:
Resolved!

Reply:
> Does this give aggregate space of all the l2cache devices? Check for the array size. spa_perf->spa_l2cache.sav_count gives the size of the sav_vdevs[] array?

sav_vdevs[] array is there..added till sav_count